docs/core/concepts/results.md

```diff
@@ -1,53 +1,44 @@
 
-## Results
+Prefect's result framework was designed with security and privacy concerns in mind. Consequently, Prefect's abstractions allow users to take advantage of all its features without ever needing to surrender control, access, or even visibility of their private data.
 
-Results represent Prefect Task outputs. In particular, anytime a Task runs, its output
-is encapsulated in a `Result` object. This object retains information about what the data is and how to "handle" it
-if it needs to be saved / retrieved at a later time (for example, if this Task requests for its outputs to be cached or checkpointed - learn more about the differences in [Persistence and Caching](persistence.md)).
+## `Result` objects
 
-An instantiated Result object has the following attributes:
-
-- a `value`: the value of a Result represents a single piece of data
-- a `safe_value`: this attribute maintains a reference to a `SafeResult` object
-  which contains a "safe" representation of the `value`; for example, the `value` of a `SafeResult`
-  might be a URI or filename pointing to where the raw data lives. In this context, "safe" means
-  that the value is safe to be sent to Cloud.
-- a `result_handler` that holds onto the `ResultHandler` used to read /
-  write the value to / from its handled representation
+A Prefect `Result` object represent data Prefect knows about, most importantly the output of a Prefect Task: anytime a Task runs, its output is encapsulated in a `Result` object. This object retains information about what the data is and how to write or read it if it needs to be saved / retrieved at a later time.
   
-In the unconfigured case, all `State` objects produced by a flow have a basic `Result` object with only `Result.value` set, and no result handler attached.
+In the unconfigured case, all `State` objects produced by a flow have a basic `Result` object with only `Result.value` set. These base objects do not know how to write themselves to persistent storage.
 
 ```python
 >>> type(state._result)
 prefect.engine.result.Result
 >>> type(state._result.value)  # this is the type of your Task's return value
 dict
->>> type(state._result.safe_value)
-prefect.engine.result.NoResultType
->>> type(state._result.result_handler)
-NoneType
 ```
 
+You can instead configure your task to use a subclass of `Result` that aligned with a persistent storage backend. This allows you to turn on persistent forms of caching and checkpointing. (Learn more about that in [Caching and Persisting Data](persistence.md). These subclasses of `Result` always have four important attributes / methods:
+
+- `location`: the location the data should be stored to (note that this can be templated as described below)
```

Comment:
Where is it described?

Reply:
@joshmeek I was going to add more extensive templating related docs after we merge this